### PR TITLE
Add STOPPED to the failure cases for Sagemaker Training Jobs

### PR DESCRIPTION
**Pull request by ferruzzi**
User pointed out that Airflow will mark a failed Sagemaker Training Job as a successful task.  Looking into the issue, Sagemaker treats "Stopped" as a failure state for this endpoint, not a successful terminal state. [Sagemaker Docs [here](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker/client/describe_training_job.html)]

To reproduce:

Run a DAG with a SagemakerTrainingOperator task.  Once the training starts, open the AWS console and stop the job.  Airflow will see that the job has ended but mark the task as successful.

After this change, those steps will show the task marked as failed as expected.

### CODE REVIEW

File: airflow/providers/amazon/aws/hooks/sagemaker.py [hunk 1]
```diff
@@ -155,6 +155,7 @@ class SageMakerHook(AwsBaseHook):
     endpoint_non_terminal_states = {"Creating", "Updating", "SystemUpdating", "RollingBack", "Deleting"}
     pipeline_non_terminal_states = {"Executing", "Stopping"}
     failed_states = {"Failed"}
+    training_failed_states = {*failed_states, "Stopped"}
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/hooks/sagemaker.py:158]: Sagemaker has so many endpoints, and more popping up so often that I am not willing to guarantee that none of them do or will ever use "Stopped" as a successful end state, so I am only adding it to the Training failed states.

**Comment by ferruzzi** [comment]
> Oh, I have so little knowledge about AWS and Sagemaker, feel afraid to judge and approve this thing :-(

It's a very large ball of wax, that's for sure.  I'm not entirely sure why you were auto-added to the reviewers list, I didn't intentionally tag you.

File: airflow/providers/amazon/aws/hooks/sagemaker.py (hunk 1 above)
Comment (o-nikolas) [reply airflow/providers/amazon/aws/hooks/sagemaker.py:158]: "Stopped" is really a failed state? What is the state called when the training job completes successfully? Is it not Stopped?

Comment (ferruzzi) [reply airflow/providers/amazon/aws/hooks/sagemaker.py:158]: `'TrainingJobStatus': 'InProgress'|'Completed'|'Failed'|'Stopping'|'Stopped',
`

"Completed" is the terminal successful state.

**Review by vincbeck — APPROVED** [review]

File: airflow/providers/amazon/aws/hooks/sagemaker.py (hunk 1 above)
Comment (o-nikolas) [reply airflow/providers/amazon/aws/hooks/sagemaker.py:158]: Got it, thanks!

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**